### PR TITLE
Fixes mech missile launchers

### LINKED ISSUE #7575: Mech Missile Rack Projectiles Do Not Fire Properly
**Problem Description**:
When you fire a missile from any* missile rack (SRM but likely others as well) it will not actually fire and instead fall onto the ground below the mech which can then be picked up and thrown as a handheld explosive

**What did you expect to happen**: 
Missiles to actually behave like missiles

**What happened instead**:
They fall down onto the floor instead

**Why is this bad/What are the consequences:**
You can use it to lug around at least three missiles on foot to throw out of a mech as well as making missile racks pointless on mechs that have them/construction

**Steps to reproduce the problem**:
Attach Missile rack to a mech/use ERT Maurader/ Mauler, raise it, fire. Cry as it lays on the floor below you

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: Anything else you can tell us.
Blob at the time, did not seem to affect it.

### PR DESCRIPTION
**Pull request by Kyep**
🆑 Kyep
fix: Mechs firing missiles no longer cause the missiles to drop at their feet, as if frozen in midair.
/🆑

Fixes #7575 

The problem was a runtime:
[02:52:45] Runtime in atoms_movable.dm,240: undefined variable /obj/mecha/combat/marauder/seraph/loaded/var/client
   proc name: throw at (/atom/movable/proc/throw_at)

It was caused by the fact the throw_at proc assumes that the thrower is something that can have a client. Rather than giving clients to mechs, I simply made it not calculate momentum for mech-launched weapons. Mech launchers are probably powered anyway, and can only face forward, so it doesn't really make sense for them to have momentum anyway.

### CODE REVIEW

File: code/game/atoms_movable.dm [hunk 1]
```diff
@@ -237,7 +237,7 @@
 		pulledby.stop_pulling()
 
 	// They are moving! Wouldn't it be cool if we calculated their momentum and added it to the throw?
-	if(thrower && thrower.last_move && thrower.client && thrower.client.move_delay >= world.time + world.tick_lag * 2)
+	if(thrower && thrower.last_move && !istype(thrower, /obj/mecha) && thrower.client && thrower.client.move_delay >= world.time + world.tick_lag * 2)
```
Comment (Fox-McCloud) [inline code/game/atoms_movable.dm]: ❄️

Comment (Kyep) [inline code/game/atoms_movable.dm]: Changed. See if you're fine with it now.

**Comment by Kyep** [comment]
@Fox-McCloud Changed. See what you think now.

**Comment by taukausanake** [comment]
That ATMTA Station reference is nice to read. Ima look more at their repo posts

**Comment by Kyep** [comment]
I tried applying Google translate to that Russian Github page.
My favorite comment:
_The fucking mug-refector is to blame for everything.
I'm not in the shower to fix it. Really. Here is the truth._
10/10, would get crosslinked again.

**Comment by ITBlackwood** [comment]
@Kyep And yet, it translated about 50% of my post correctly. Almost. Kinda.

**Comment by taukausanake** [comment]
@Kyep That's EXACTLY WHY I want to look through it more

**Comment by outbools** [comment]
_That's all fucking move refactor fault
I dont have a fucking clue how to fix it. Seriously. I'm telling the truth._

**Comment by outbools** [comment]
@taukausanake what we did on our server:
1)Karma-locked all species except human
2)Refactored karmashop to buy some immersion-breaking clothes as well
3)Refactored karma. You can decrease someone's karma by 1, you can get karma for being successful antags, admins can give any amount of karma to specified ckey. (there was an idea to finish medals and give karma for medals but im too lazy to implement it)
4)Our server is whitelisted. Non-whitelisted can spawn as a D-class personnel, currently they have zero use and spawning in permabrig with prisoner clothes. There was a lot of prison breaks, riots and other funny things going on. Now it's nerfed to being almost impossible to break free.
5)Remaped cyberiad a little bit
6)we got some minor stuff from tg like attorneys badges, ARCD, also variety types of glass - wine glass, champagne glass, etc

File: code/game/atoms_movable.dm [hunk 2]
```diff
@@ -237,7 +237,7 @@
 		pulledby.stop_pulling()
 
 	// They are moving! Wouldn't it be cool if we calculated their momentum and added it to the throw?
-	if(thrower && thrower.last_move && thrower.client && thrower.client.move_delay >= world.time + world.tick_lag * 2)
+	if(thrower && thrower.last_move && isliving(thrower) && thrower.client && thrower.client.move_delay >= world.time + world.tick_lag * 2)
```
Comment (Fox-McCloud) [inline code/game/atoms_movable.dm]: This is adding sanity where it's not needed, still.

Look at the *code* and how it's parsed and what its arguments are and how they're used to determine a problem; if you do that, then it makes it easier to debug things and you can avoid adding performance overhead or checks like this where they're *not necessary*.

You shouldn't be attempting to pass a mob argument to `thrower` in the first place.

**Comment by taukausanake** [comment]
@outbools I want allllll of that

**Comment by outbools** [comment]
@taukausanake you can always cherry-pick it 😏

File: code/game/atoms_movable.dm (hunk 2 above)
Comment (Kyep) [inline code/game/atoms_movable.dm]: Changed.

**Comment by Kyep** [comment]
@Fox-McCloud Updated with new fix method.

**Comment by Fox-McCloud** [comment]
Thank you.

**Pull request merged**